core/server/worker/src/main/java/alluxio/worker/block/TieredBlockStore.java

```diff
@@ -500,46 +495,32 @@ private void checkTempBlockIdAvailable(long blockId) throws BlockAlreadyExistsEx
    *
    * @param sessionId the id of session
    * @param blockId the id of block
-   * @throws BlockDoesNotExistException if block id can not be found in temporary blocks
-   * @throws BlockAlreadyExistsException if block id already exists in committed blocks
-   * @throws InvalidWorkerStateException if block id is not owned by session id
    */
-  private void checkTempBlockOwnedBySession(long sessionId, long blockId)
-      throws BlockDoesNotExistException, BlockAlreadyExistsException, InvalidWorkerStateException {
-    if (mMetaManager.hasBlockMeta(blockId)) {
-      throw new BlockAlreadyExistsException(ExceptionMessage.TEMP_BLOCK_ID_COMMITTED, blockId);
-    }
-    TempBlockMeta tempBlockMeta = mMetaManager.getTempBlockMeta(blockId);
-    long ownerSessionId = tempBlockMeta.getSessionId();
-    if (ownerSessionId != sessionId) {
-      throw new InvalidWorkerStateException(ExceptionMessage.BLOCK_ID_FOR_DIFFERENT_SESSION,
-          blockId, ownerSessionId, sessionId);
+  private TempBlockMeta checkAndGetTempBlockMeta(long sessionId, long blockId) {
+    Optional<TempBlockMeta> tempBlockMeta;
+    try (LockResource r = new LockResource(mMetadataReadLock)) {
+      tempBlockMeta = mMetaManager.getTempBlockMeta(blockId);
     }
+    checkState(!mMetaManager.hasBlockMeta(blockId)
+        && tempBlockMeta.isPresent()
+        &&tempBlockMeta.get().getSessionId() == sessionId,
+        format("Session %d doesn't own block %d", sessionId, blockId));
```

Comment:
I think it's a good idea to aggregate those exceptions but might be more insightful to provide which one goes wrong when debugging